src/components/organisms/SearchPane/CreateFilteredNode.tsx

```diff
@@ -0,0 +1,61 @@
+import parse from 'html-react-parser';
+
+import {FileEntry, MatchNode} from '@models/fileentry';
+
+import {FilterTreeNode} from '../FileTreePane/types';
+
+import * as S from './styled';
+
+type Props = {
+  item: FileEntry;
+};
+
+type MatchLineProps = {
+  lineMatches: MatchNode[];
+};
+
+export const createFilteredNode = (filteredFileMap: FileEntry[]): FilterTreeNode => {
+  const Title = ({item}: Props) => (
+    <S.NodeContainer>
+      <S.NodeTitleContainer>
+        <S.EntryName className="file-entry-name">{item.name}</S.EntryName>
+        <S.MatchCount>{item.matchCount}</S.MatchCount>
+        <S.Path className="file-entry-path">{item.filePath}</S.Path>
+      </S.NodeTitleContainer>
+    </S.NodeContainer>
+  );
+
+  const StyledMatchLine = ({lineMatches}: MatchLineProps) => {
+    const wholeLine = lineMatches.reduce((acc: string, lm: MatchNode) => acc + lm.textWithHighlights, '');
+    return <S.MatchLine>{parse(wholeLine)}</S.MatchLine>;
```

Comment:
because we have <span> tag for css styling (yellow background in tree pane) to wrap around matched query;
React escapes all tags so it's either dangerously set HTML or a parser library

Reply:
it's still not clear to me why we need this, could there be a solution with CSS only? I'm hesitant on using the `html-react-parser` or the dangerously set HTML

Reply:
@devcatalin, I need to style some specific part of the string which I wrap in a tag, in this case <em> tag. So far I don't see another way how I can embed it in the code without using html-react-parser or dangerouslySetHtml since the tag will appear as a plain text and not as a tag. Maybe I'm missing another solution for it... Any thoughts?